File: backend/app/schemas/public.py
from datetime import date

from pydantic import BaseModel


class PublicServisTerakhir(BaseModel):
    tanggal: date
    jenis_pekerjaan: str | None = None


class PublicArmada(BaseModel):
    kode_armada: str
    jenis: str | None = None
    merk_type: str | None = None
    no_polisi: str | None = None
    status_armada: str
    foto_url: str | None = None
    servis_terakhir: PublicServisTerakhir | None = None

File: frontend/src/types/public.ts
export interface PublicServisTerakhir {
  tanggal: string;
  jenis_pekerjaan: string | null;
}

export interface PublicArmada {
  kode_armada: string;
  jenis: string | null;
  merk_type: string | null;
  no_polisi: string | null;
  status_armada: string;
  foto_url: string | null;
  servis_terakhir: PublicServisTerakhir | null;
}
